src/Backups/RestoreCoordinationDistributed.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <Backups/IRestoreCoordination.h>
+#include <Common/ZooKeeper/Common.h>
+#include <map>
+#include <mutex>
+
+
+namespace DB
+{
+
+/// Stores restore temporary information in Zookeeper, used to perform RESTORE ON CLUSTER.
+class RestoreCoordinationDistributed : public IRestoreCoordination
+{
+public:
+    RestoreCoordinationDistributed(const String & zookeeper_path_, zkutil::GetZooKeeper get_zookeeper_);
+    ~RestoreCoordinationDistributed() override;
+
+    void setOrGetPathInBackupForZkPath(const String & zk_path_, String & path_in_backup_) override;
+
+    bool acquireZkPathAndName(const String & zk_path_, const String & name_) override;
+    void setResultForZkPathAndName(const String & zk_path_, const String & name_, Result res_) override;
+    bool getResultForZkPathAndName(const String & zk_path_, const String & name_, Result & res_, std::chrono::milliseconds timeout_) const override;
```

Comment:
Unclear semantics -- why do we have both `res_` and `bool` as return value?

Reply:
The function returns the result if it's ready or wait for it for a specified amount of time.
So the function returns true if it returned the result or returns false if the time is out.